### DISCUSSION TITLE
Возможно ли видеть изменения в Totum, не используя клавишу F5 ?

### DISCUSSION THREAD

**Discussion by IrinaG1090** [totumonline/totum-ru-issues-and-discussions#480] [Как это сделать?] [answered]
1)  При внесении изменений в таблицу, строку или ячейку USER-ом 1, эти изменения видны для USER-а 2 только после обновления страницы. 
Есть ли такая возможность чтоб видеть изменения сразу без обновления страницы? 

2) Как настроить таблицу(простую или цикл) чтоб при загрузке ТОТУМ (после входа в систему ) она первая прогружалась на начальной странице по умолчанию? 

(сейчас работает так: выбираю папку - выбираю таблицу) то есть 2 раза клацаю мышью. Можно ли это обойти?

Благодарю за ответы

**Comment by ttm-alexeypolunin** [comment, marked as answer]
1. PRO — Автообновление таблиц https://ru.docs.totum.online/tables#table-settings-actual

**Автообновление:** При срабатывании нотификации о изменении таблицы другим пользователем автоматически обновляет данные на странице. Если в этот момент у пользователя будет открыто поле на редактирование, панель или окно — автоматического обновления не произойдет, будет показана стандартная нотификация о изменении в фоне.

PRO Автообновление — только в PRO, аналогично Нестрогое + Автообновление

! PRO Без отслеживания с автообновлением — только в PRO, аналогично ! Без отслеживания изменений + Автообновление

> Не стоит это включать если у вас сервер — это плата от микроволновки, а с таблицей работает одновременно 30 человек!

2. В таблице ролей поле Таблица редиректа или как-то так

**Важная инфа:** вот эта таблица редиректа при переходе на 5 версию будет переведена в PRO вместе с другим функционалом, а PRO будет оснащен системой лицензирования. Первые 10 пользователей в нем будут бесплатные — остальное за деньги.

Все желающие смогут конечно не обновляться и остаться на 4, но она не будет получать обновления ошибок и техподдержку тк у нас на это нет ресурсов — мы будем концентрироваться на 5 коммерческой версии.

**Comment by k1erk** [comment]
> **Важная инфа:** вот эта таблица редиректа при переходе на 5 версию будет переведена в PRO вместе с другим функционалом, а PRO будет оснащен системой лицензирования. Первые 10 пользователей в нем будут бесплатные — остальное за деньги.
> 
А можно немного прояснить что значит 10 пользователей и как это будет работать ?

**Reply by ttm-alexeypolunin** [reply]
PRO будет изменен и у него появится компилированное ядро на GO, которое в оперативке будет хранить информацию о пользователях и полях/таблицах. Это нужно, что бы улучшить работоспособность систем с большим одновременным онлайном, тк эти данные читаются при каждом действии и из ОЗУ это делать быстрее. Оно же будет по web socket отвечать на запросы по нотификациям, которые потребляют воркеров fpm. Вместо fpm будут множится GO-рутины. Ну и прочих ништяков там можно будет сделать. Например: крон запускающийся сразу как выполнился предыдущий, или показ процессов и их точек старта в онлайн для поиска нагрузки и тп.

Этот же модуль будет обрабатывать лицензии.

Первые 10 пользователей останутся бесплатно (всего 13 тк есть 3 технических). При превышении количества и отсутствии лицензии система будет блокироваться на изменение тк права доступа будут обрабатываться в GO.

Лицензия будет покупаться на количество пользователей на период. Цены 2-6 доллара за пользователя сверх лимита в месяц. Для оплаты из остального мира будет подороже.

Помимо этого, мы уберем большое количество функционала из MIT:

— подключение внешних функций
— tree-view
— поворот таблицы
— анонимные
— формы
— карточки и канбан
— json-api
**— файлы**
— таблица редиректа

Так как он слишком хороший.

Всегда остается возможность **не обновляться**. Для тех, кто решит остаться на 4 версии с нами случилось то, что обычно случается с опенсорсными проектами без финансирования — разрабы ушли работать в FAANG и перестали проект поддерживать. Я думаю 98% проектов не будут переходить на 5 версию 🤷‍♀️

Также отсутствие простой системы оплаты тормозит развитие как с точки зрения того, что мы не чиним ошибки, копятся пробелы в документации, недостаточно занимаемся обучением GPT писать код на тотум и еще много чем. Так и с точки зрения стартующих крупные проекты. Таких пользователей очень интересует, не загнется ли проект завтра, а когда нет понятной коммерческой составляющей — точно загнется.

Все дополнительные способы комерциализации типа допсервисов, обучения, разработки и облака — все это туфта и не может обеспечить работу двух человек даже с зарплатой доставщика пиццы.

У нас осталось немного денег на следующий год, вот мы к нему и готовимся.

**Reply by ttm-alexeypolunin** [reply]
**Уточнения:**

2-6 по принципу «чем больше пользователей, тем дешевле». Те на проектах с большим количеством пользователей будет минимальная цена, а на совсем больших проектах — **спеццена ниже минимальной.**

На одну лицензию будет привязываться 2 хоста (рабочий и тестовый). Но разницы между ними не будет — те это могут быть 2 работающих проекта.

Всем, кто пользовался PRO (поставил звезду на GH и получил доступ к PRO-репо) — будем предлагать большие скидки на переход на V5 в первые 3 месяца наверное (может и дольше). Оплатить со скидкой можно будет любой период.

**Reply by vldekov** [reply]
10 пользователей в таблице или речь идет об одновременно активных пользователях ( подключения ) ?

**Reply by ttm-alexeypolunin** [reply]
10 активных пользователей в таблице users

**Reply by vldekov** [reply]
Если одним логином пользуются 5 человек, то тарифицироваться будет как один пользователь?

**Reply by ttm-alexeypolunin** [reply]
Конечно, как мы их различим - никак.

**Reply by vldekov** [reply]
https://ru.totum.online/cloud
Эти тарифы будут действовать для v5 ?

**Reply by ttm-alexeypolunin** [reply]
Для нас это облачное размещение сильно вторично. Я думаю, что там также добавится за количество пользователей. И возможно мы переформатируем последний тариф. Но это такое - это облако нафиг никому не нужно, ни пользователям, ни нам.

**Reply by ttm-alexeypolunin** [reply]
> Конечно, как мы их различим - никак.

Подумали и придумали как различать. 5 одновременных подключений (разных браузеров) от одной учетки будут потреблять 5 слотов лицензии.

**Reply by LostAlly** [reply]
> > Конечно, как мы их различим - никак.
> 
> Подумали и придумали как различать. 5 одновременных подключений (разных браузеров) от одной учетки будут потреблять 5 слотов лицензии.

Как будет отрабатываться ситуация с переходом с места на место?

Вообще очень не хватает быстрого переключения между пользователями и временной блокировки с входом по «пинкоду» быстрому. Т.е. залогинился на одном месте например три человека и возможность быстрого переключения между ними по короткому паролю. И возможность временной блокировки если сотрудник отходит от рабочего места и есть вероятность, что может придти другой.

На самом деле данную ситуацию можно решить несколькими браузерами, но быстрая блокировка и в этом случае очень бы пригодилась.

) и когда будут новости по новой версии? )

**Reply by erpic-ru** [reply]
Ну вообще, если сильно хочется, то при помощи этой штуки можно собрать что-то с быстрой авторизацией:
https://ru.docs.totum.online/auth-links

**Reply by ttm-alexeypolunin** [reply]
Блокируются экраны на обеих местах с показом кнопки «Продолжить здесь». При нажатии в одном из браузеров — разлогин на всех остальных.

В PRO есть авторизационные ссылки https://ru.docs.totum.online/auth-links, можете написать любую логику. Вы даже кнопку можете в меню пользователя сделать для переключения, добавив ее в h_user_settings_buttons и поставив его как удерживаемое поле https://ru.docs.totum.online/update#held-fields (чтоб не стерлось при обновлении). Мы там точно не будем дополнительно наворачивать, тк и без этого капец уже навернуто сверх необходимого.

Временная блокировка — это разлогин. Если вам нужно быстрый разлогин — уменьшаете время сессии (но все жаловались, что стандартные 30 минут это очень коротко, поэтому она там теперь 2 часа)

Сентябрь.

**Reply by ttm-alexeypolunin** [reply]
Будет настройка, которая позволит задать количество одновременных сессий для одного логина, тк есть случаи, когда удобнее спроектировать систему для работы нескольких человек под одним логином. В этом случае при превышении не знаю кого будем разлогинивать — сессию с наименьшим временем создания... не знаю пока
